src/main/java/uk/gov/hmcts/ccd/data/casedetails/CachedCaseDetailsRepository.java

```diff
@@ -43,29 +44,52 @@ public CaseDetails set(final CaseDetails caseDetails) {
         return caseDetailsRepository.set(caseDetails);
     }
 
+    @Override
+    public Optional<CaseDetails> findById(String jurisdiction, Long id) {
+        final Function<Long, CaseDetails> findCase = (key) -> caseDetailsRepository.findById(jurisdiction, id)
+                                                                                   .orElse(null);
+        return Optional.ofNullable(idToCaseDetails.computeIfAbsent(id, findCase))
+                       .filter(caseDetails -> jurisdiction.equals(caseDetails.getJurisdiction()));
```

Comment:
if we are doing a case detail find by id and jurisdiction why are we performing a filter on jurisdiction here?

Reply:
The reason for the filter is because the same cache is used by both `findById(id)` and `findById(jurisdiction, id)`. Because of that, the cache could be populated by `findById(id)` which doesn't enforce jurisdiction but `findById(jurisdiction, id)` is expected to enforce jurisdiction check.

I agree that this seems like a trivial case and adds complexity; so I'm removing that filter.

Reply:
Changed cache to `Map<Long, Optional<CaseDetails>>`